src/common/commonutils/CommandUtils.c

```diff
@@ -29,270 +31,200 @@ static int NormalizeStatus(int status)
     return newStatus;
 }
 
-static int SystemCommand(void* context, const char* command, int timeoutSeconds, CommandCallback callback, OsConfigLogHandle log)
-{
-    const int callbackIntervalSeconds = 5; //seconds
-    const int defaultCommandTimeout = 60; //seconds
+#define BUFFER_SIZE 1024
 
-    pid_t intermediateProcess = -1;
-    pid_t workerProcess = -1;
-    pid_t timerProcess = -1;
-    pid_t childProcess = -1;
-    int status = -1;
-    int intermediateStatus = -1;
-    int totalWaitSeconds = 0;
-    int timeout = (timeoutSeconds > 0) ? timeoutSeconds : defaultCommandTimeout;
+int ExecuteCommand(void* context, const char* command, bool replaceEol, bool forJson, unsigned int maxTextResultBytes, unsigned int timeoutSeconds,
+    char** textResult, CommandCallback callback, OsConfigLogHandle log)
+{
+    int workerPid = -1;
+    int pipefd[2] = {0};
 
-    bool mainProcessThread = (bool)(getpid() == gettid());
+    if (NULL == command)
+    {
+        OsConfigLogDebug(log, "Command cannot be NULL");
+        return -1;
+    }
+    // Create a pipe, then fork. Forked process dup2()s the write pipe end to stdout and stderr,
+    // then execl()s the shell with the given command. The main process uses timeoutable select()
+    // to read from the pipe, and keep track of both command  timeout and callbacks. The read loop
+    // ends when the read() returns EOF.
 
-    fflush(NULL);
+    if (0 != pipe(pipefd))
+    {
+        OsConfigLogError(log, "Cannot create pipe for command '%s' - %s", command, strerror(errno));
```

Comment:
>command '%s' - %s" [](http://example.com/codeflow?start=54&length=18)

nit This ( '-')is a log pattern that is not used in the existing code of this project. 

Instead of this:

```
OsConfigLogError(log, "Cannot create pipe for command '%s' - %s", command, strerror(errno));
````

Can this new version keep the old version's trace style, for example:

```
OsConfigLogError(log, "Cannot create pipe for command '%s', pipe() failed with %d ('%s')", command, errno, strerror(errno));
````
Here and elsewhere.

Reply:
Done